modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java

```diff
@@ -228,7 +228,7 @@ public void setAgentSetting(String key, String value) {
             return new Setting<>(qualifiedKey, "", (value) -> {
                 if (qualifiedKey.equals("_na_") == false && PERMITTED_AGENT_KEYS.contains(key) == false) {
                     // TODO figure out why those settings are kept, these should be reformatted / removed by now
-                    if (key.startsWith("global_labels.")) {
```

Comment:
Doh, `key` extraction is also somehow flawed here, it only ever includes the last part :/